src/UserContext/RoleProvider.php

```diff
@@ -52,10 +54,15 @@ public function updateUserContext(UserContext $context)
             throw new InvalidConfigurationException('The context hash URL must be under a firewall.');
         }
 
-        if (null === $token = $this->tokenStorage->getToken()) {
+        $token = $this->tokenStorage->getToken();
+        if (null === $token && class_exists(AnonymousToken::class)) {
             return;
         }
 
+        if (null === $token) {
+            $token = new NullToken();
```

Comment:
what is the purpose of instantiating a NullToken here? if token storage does not give us that null token, should we not avoid updating the context?

i think we should add a comment why we do this, and one above to explain until what symfony version this will be the case.

Reply:
Yes I was not totally sure of this solutation as the AnonymousToken is gone and the token storage just returns null if your not logged in. But the null token is like a replacement for the AnonymousToken by I think it's only used by symfony in the voter system so that you actually always receive a token.

I did the fix like this because it would fix the tests and seemed logic to me. As previously the AnonymousToken would return an empty array of user roles that was used in the calculation of the hash, with the new behaviour this was skipped and I only could get it to work with the null token. So we might have to look into this if it's the correct fix.